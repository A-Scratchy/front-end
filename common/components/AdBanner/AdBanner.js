import React from 'react';
import PropTypes from 'prop-types';
import OutboundLink from 'common/components/OutboundLink/OutboundLink';
import styles from './AdBanner.css';

AdBanner.propTypes = {
  altText: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  imageSource: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};

function AdBanner({
  altText, children, imageSource, link,
}) {
  return (
    <OutboundLink
      href={link}
      analyticsEventLabel={`[AdBanner Hit] to ${link}`}
      className={styles.adBannerLink}
    >
      <div className={styles.adBanner}>
        <div className={styles.adBannerImageContainer}>
          <img
            alt={altText}
            className={styles.adBannerImage}
            src={imageSource}
          />
        </div>
        <h4 className={styles.adBannerText}>{children}</h4>
      </div>
    </OutboundLink>
  );
}

export default AdBanner;
