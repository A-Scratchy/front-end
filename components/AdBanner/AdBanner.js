import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
import Router from 'next/router';
import styles from './AdBanner.css';

AdBanner.propTypes = {
  altText: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  href: PropTypes.string.isRequired,
  imageSource: PropTypes.string.isRequired,
};

AdBanner.defaultProps = { className: undefined };

function AdBanner({ altText, children, className, imageSource, href }) {
  return (
    <OutboundLink
      analyticsEventLabel="[AdBanner Hit]"
      className={classNames(className, styles.adBannerLink)}
      hasIcon={false}
      href={href}
      router={Router}
    >
      <div className={styles.adBanner}>
        <div className={styles.adBannerImageContainer}>
          <img alt={altText} className={styles.adBannerImage} src={imageSource} />
        </div>
        <h4 className={styles.adBannerText}>{children}</h4>
      </div>
    </OutboundLink>
  );
}

export default AdBanner;
