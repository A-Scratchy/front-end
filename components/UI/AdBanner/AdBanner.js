import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import OutboundLink from 'common/components/OutboundLink/OutboundLink';
import styles from './AdBanner.css';

AdBanner.propTypes = {
  altText: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  href: PropTypes.string.isRequired,
  imageSource: PropTypes.string.isRequired,
};

AdBanner.defaultProps = {
  className: undefined,
};

function AdBanner({
  altText, children, className, imageSource, href,
}) {
  return (
    <OutboundLink
      analyticsEventLabel={`[AdBanner Hit] to ${href}`}
      className={classNames(className, styles.adBannerLink)}
      hasIcon={false}
      href={href}
    >
      <div className={styles.adBanner}>
        <div className={styles.adBannerImageContainer}>
          <img
            alt={altText}
            className={styles.adBannerImage}
            src={imageSource}
          />
        </div>
        <h4 className={styles.adBannerText}>{children}</h4>
      </div>
    </OutboundLink>
  );
}

export default AdBanner;
