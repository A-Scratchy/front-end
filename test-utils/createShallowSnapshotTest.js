/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
