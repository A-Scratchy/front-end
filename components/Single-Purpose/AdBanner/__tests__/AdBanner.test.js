import React from "react";
import createShallowSnapshotTest from "test-utils/createShallowSnapshotTest";

import AdBanner from "../AdBanner";

describe('AdBanner', () => {
  test("it should render properly with required props", () => {
    createShallowSnapshotTest(
        <AdBanner 
            altText="Some text" 
            href="www.test.com"
            imageSource="www.imagetest.com">
            Test
        </AdBanner>
    );
  });
});
