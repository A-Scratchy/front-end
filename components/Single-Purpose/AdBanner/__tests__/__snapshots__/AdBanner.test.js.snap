// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdBanner it should render properly with required props 1`] = `
<OutboundLink
  analyticsEventLabel="[AdBanner Hit] to www.test.com"
  className="adBannerLink"
  hasIcon={false}
  href="www.test.com"
>
  <div
    className="adBanner"
  >
    <div
      className="adBannerImageContainer"
    >
      <img
        alt="Some text"
        className="adBannerImage"
        src="www.imagetest.com"
      />
    </div>
    <h4
      className="adBannerText"
    >
      Test
    </h4>
  </div>
</OutboundLink>
`;
